resources/js/Components/Modal/EventRegistrationModal.tsx

```diff
@@ -0,0 +1,165 @@
+            ? registrationData.characters[0].id
+            : undefined,
+    }
+
+    const {
+        data,
+        setData,
+        post,
+        errors,
+        clearErrors,
+        delete: destroy,
+        wasSuccessful,
+        processing,
+    } = useForm<FormDataType>(SESSION_REGISTRATION_FORM_INITIAL_VALUE)
+
+    useEffect(() => {
+        if (wasSuccessful) {
+            clearErrors()
+            onClose()
+        }
+    }, [wasSuccessful])
+
+    const {t} = useTranslation()
+
+    return (
+        <Modal
+            open={isRegisterModalOpen}
+            onClose={onClose}
+            closeAfterTransition
+            BackdropComponent={Backdrop}
+            BackdropProps={{
+                timeout: 500,
+            }}>
+            <Fade in={isRegisterModalOpen}>
+                <StyledBox borderRadius={2}>
+                    <Grid container>
+                        <Grid item md={12}>
+                            <StyledTypography>
+                                {registrationData.is_registered
+                                    ? t('eventDetail.leave-message')
+                                    : t('eventDetail.registration-message')}
+                            </StyledTypography>
+                        </Grid>
+                        {!registrationData.is_registered && (
+                            <Grid item xs={12}>
+                                <Select
+                                    id='character_id'
+                                    label={t('form.chosen-character')}
+                                    name='Chosen Character'
+                                    value={data.character_id}
+                                    required
+                                    onChange={(e) =>
+                                        setData('character_id', parseInt(e.target.value))
+                                    }
+                                    options={allUserCharacters}
+                                />
+                                {errors && (
+                                    <ErrorText
+                                        message={
+                                            errors?.character_id ||
+                                            errors?.overlap ||
+                                            errors?.seats
+                                        }
+                                    />
+                                )}
+                            </Grid>
+                        )}
+                        <Grid item md={6}>
+                            {registrationData.is_registered ? (
+                                <Button
+                                    loading={processing}
+                                    variant='contained'
+                                    fullWidth
+                                    onClick={() => {
+                                        if (data.session_id) {
```

Comment:
this if statement is not needed anymore

Reply:
I'm still getting an eslint error on this. I might keep it.